gateway/hail.nginx.conf.in

```diff
@@ -90,80 +90,92 @@ server {
+            add_header 'Content-Length' 0;
+            return 204;
+        }
+
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Allow-Credentials' 'true';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, DELETE, OPTIONS';
+        add_header 'Access-Control-Allow-Headers' 'Authorization,Keep-Alive,User-Agent,X-Requested-With,Cache-Control,Content-Type';
+
+        auth_request /auth;
+        auth_request_set $auth_user $upstream_http_user;
+        auth_request_set $auth_scope $upstream_http_scope;
+        proxy_pass http://notebook;
         proxy_set_header Host $http_host;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Host $http_host;
         proxy_set_header X-Forwarded-Proto $scheme;
         proxy_set_header X-Real-IP $remote_addr;
-        proxy_http_version 1.1;
-        proxy_redirect off;
-        proxy_buffering off;
         proxy_set_header Upgrade $http_upgrade;
         proxy_set_header Connection "upgrade";
-        proxy_read_timeout 86400;
+        proxy_set_header User $auth_user;
+        proxy_set_header Scope $auth_scope;
 
-        proxy_connect_timeout 5s;
-
-        proxy_intercept_errors on;
-        # 502: no service, 504: pod not responding
-        error_page 502 504 = @pod_dead;
+        proxy_http_version 1.1;
+        proxy_read_timeout 86400;
+        proxy_buffering off;
     }
 
-    location @pod_dead {
-        return 307 $scheme://$http_host/new;
-    }
+    location ~ /instance/([^/]+)/(.*) {
+        set $svc_name $1;
+        auth_request /auth-notebook;
 
-    location ~ /instance-ready/([^/]+)/ {
         resolver kube-dns.kube-system.svc.cluster.local;
-        proxy_pass http://$1.default.svc.cluster.local/instance/$1/;
-    }
-
-    location / {
-        proxy_pass http://notebook/;
+        proxy_pass http://$svc_name.default.svc.cluster.local$request_uri;
 
         proxy_set_header Host $http_host;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Host $http_host;
         proxy_set_header X-Forwarded-Proto $scheme;
         proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
         proxy_http_version 1.1;
         proxy_redirect off;
         proxy_buffering off;
-        proxy_set_header Upgrade $http_upgrade;
-        proxy_set_header Connection "upgrade";
         proxy_read_timeout 86400;
-    }
-
-    listen [::]:443 ssl;
-    listen 443 ssl;
-    ssl_certificate /etc/letsencrypt/fullchain.pem;
-    ssl_certificate_key /etc/letsencrypt/privkey.pem;
-    include /etc/letsencrypt/options-ssl-nginx.conf;
-    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
-}
-
-server {
-    server_name app.@domain@;
-
-    location / {
-        proxy_pass http://web/;
+        proxy_connect_timeout 10s;
```

Comment:
None really. I was hoping 10s would help us deal with dns prop issues, but it just waits 10s to die, even if propagation occurred within that span. Annoying! 

I kind of think 5s, or 10s, or 60s doesn't matter much. I'm happy to set it to 75s, to prevent it from ever being something to think about on the gateway (or I guess later this week on the notebook nginx side), and our job will be to make sure users never wait anything resembling 5s to connect, unless the nuclear apocalypse happens and we're back on 28.8 baud modems (if that?)